Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -332,6 +335,27 @@ static ThrowCompletionOr<Value> regexp_builtin_exec(VM& vm, RegExpObject& regexp
         }
     }
 
+    // https://github.com/tc39/proposal-regexp-legacy-features#regexpbuiltinexec--r-s-
+    // 5. Let thisRealm be the current Realm Record.
+    auto* this_realm = &realm;
+    // 6. Let rRealm be the value of R's [[Realm]] internal slot.
+    auto* rRealm = regexp_object.vm().current_realm();
```

Comment:
This is not correct, the VM's current realm is something entirely different. You'll need to add a `NonnullGCPtr<Realm>` member to `RegExpObject` and implement `// FIXME: 3. Set the value of obj’s [[Realm]] internal slot to thisRealm.` from above